src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -524,6 +533,14 @@ private function validateAndDenormalize(array $types, string $currentClass, stri
                 return (float) $data;
             }
 
+            if (($context[self::ENABLE_STRING_BOOL_VALUE] ?? $this->defaultContext[self::ENABLE_STRING_BOOL_VALUE] ?? false) && Type::BUILTIN_TYPE_BOOL === $builtinType && \is_string($data)) {
+                if (null !== $booleanValue = filter_var($data, \FILTER_VALIDATE_BOOL, \FILTER_NULL_ON_FAILURE)) {
+                    return $booleanValue;
+                }
+
+                throw new NotNormalizableValueException(sprintf('The value of the "%s" attribute for class "%s" must be one of "true", "yes", "on", "1", "false", "no", "off", "0", "" ("%s" given).', $attribute, $currentClass, $data));
```

Comment:
```suggestion
                throw new NotNormalizableValueException(sprintf('The value of the "%s" attribute for class "%s" must be one of "true", "yes", "on", "1", "false", "no", "off", "0", ""(empty string) ("%s" given).', $attribute, $currentClass, $data));
```

Not sure if this could help and if it makes sense

Reply:
Maybe, I was trying to follow Symfony style from exceptions around.